examples_source/2D_simulation(crystalline)/plot_10_COSY.py

```diff
@@ -0,0 +1,179 @@
+from mrsimulator import Method, SpectralDimension
+from mrsimulator.method import SpectralEvent, RotationEvent
+from mrsimulator.spin_system.isotope import Isotope
+from mrsimulator import Site, Coupling, SpinSystem
+from mrsimulator import signal_processor as sp
+
+import matplotlib.pyplot as plt
+
+
+# %% [markdown]
+# Generate the site, coupling, and spin system objects.
+
+# %%
+site_A = Site(isotropic_chemical_shift=2.0)
+site_X = Site(isotropic_chemical_shift=3.0)
+
+coupling_AX = Coupling(site_index=[0, 1], isotropic_j=20)
+
+ax_system = SpinSystem(sites=[site_A, site_X], couplings=[coupling_AX])
+
+# %% [markdown]
+# Create methods for the path and antipath of the COSY experiment
+
+# %%
+p_plus_query = [{"ch1": {"P": [+1]}}]
+p_minus_query = [{"ch1": {"P": [-1]}}]
+trans_queries = [
+    p_minus_query,
+    p_plus_query,
+]
+
+# Simulate a H-1 spectrum at 300 MHz |\label{ln:H1}|
+H1 = Isotope(symbol="1H")
+H1_ref_freq = 300  # MHz
+B0 = H1.ref_freq_to_B0(H1_ref_freq * 1e6)  #
+# Set the spectral width and reference offset |\label{ln:H1sw}|
+sw = 1.5 * H1_ref_freq
+offset = 2.5 * H1_ref_freq
+offset_sign = [1, -1]
+
+methods = [
+    Method(
+        channels=["1H"],
+        magnetic_flux_density=B0,
+        spectral_dimensions=[
+            SpectralDimension(
+                count=2048,
+                spectral_width=sw,  # in Hz
+                reference_offset=o_sgn * offset,  # in Hz
+                events=[
+                    SpectralEvent(transition_queries=t_queries),
+                    RotationEvent(ch1=Rotation(angle=np.pi / 2, phase=-np.pi / 2)),
+                ],
+            ),
+            SpectralDimension(
+                count=2048,
+                spectral_width=sw,  # in Hz
+                reference_offset=offset,  # in Hz
+                events=[SpectralEvent(transition_queries=[{"ch1": {"P": [-1]}}])],
+            ),
+        ],
+    )
+    for o_sgn, t_queries in zip(offset_sign, trans_queries)
+]
+
+# %% [markdown]
+# Create the Simulator object with the spin system and two methods, and run the
+# simulation.
+
+# %%
+sim = Simulator(spin_systems=[ax_system], methods=methods)
+sim.config.integration_density = 1
+sim.config.number_of_sidebands = 1
+sim.run()
+
+# %% [markdown]
+# Build the processor, and apply the broadening and tophat apodization to all datasets.
+# Perform hypercomplex processing by flipping the antipath and adding it to the path
+# signal.
+
+# %%
+proc = sp.SignalProcessor(
+    operations=[
+        sp.IFFT(dim_index=(0, 1)),
+        sp.apodization.TopHat(rising_edge="0 s", dim_index=0),
+        sp.apodization.TopHat(rising_edge="0 s", dim_index=1),
+        sp.apodization.Exponential(FWHM="1 Hz", dim_index=0),
+        sp.apodization.Exponential(FWHM="1 Hz", dim_index=1),
+        sp.FFT(dim_index=(0, 1)),
+    ]
+)
+
+processed_path = proc.apply_operations(dataset=sim.methods[0].simulation)
+processed_antipath = proc.apply_operations(dataset=sim.methods[1].simulation)

```

Comment:
isn't method at index 0 antipath (-1 -> -1) and at 1 is path  (+1 -> -1)?

Reply:
No, path is  -1,-1 and antipath is +1,-1, see slide 32 at this link: https://grandinetti.org/resources/2024-NMRSequenceDesign.pdf

Reply:
also, when I flip them, the nu_1 dimension goes negative, so it doesn't look correct in the simulation either